apps/client/src/register/components/registerForm.tsx

```diff
@@ -0,0 +1,143 @@
+import { Button } from '@mui/material';
+import FormInput from '../../components/FormInput';
+import React, { useState } from 'react';
+import axios from 'axios';
+
+const RegisterForm: React.FC = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const validatePassword = (password: string) => {
+    if (password.length < 6) {
+      setPasswordError('Password must be at least 6 characters long');
+      return false;
+    } else {
+      setPasswordError('');
+      return true;
+    }
+  };
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newPassword = e.target.value;
+    setPassword(newPassword);
+    validatePassword(newPassword);
+    // Check if confirm password matches the new password
+    if (confirmPassword && confirmPassword !== newPassword) {
+      setConfirmPasswordError('Passwords do not match');
+    } else {
+      setConfirmPasswordError('');
+    }
+  };
+
+  const handleConfirmPasswordChange = (
```

Comment:
this type of functions should go in a separate file, and make them pure, so they don't receive an event, just the string, also add tests